src/bp/ui-studio/src/web/components/Shared/ElementsList/InputElement.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import classnames from 'classnames'
+import style from './style.scss'
+
+export class InputElement extends React.Component {
+  elementInputRef = React.createRef()
+
+  state = {
+    error: undefined
+  }
+
+  onEnter = value => {
+    this.props.onEnter(value)
+  }
+
+  handleOnEnter = event => {
+    const inputValue = event.target.value
+    const enterKeyPressed = event.keyCode === 13
+    const inputNotEmpty = inputValue !== ''
+    const elementUnique = !this.props.elements.includes(inputValue)
+
+    if (enterKeyPressed) {
+      if (inputNotEmpty && elementUnique) {
+        this.setState({ error: undefined }, () => this.onEnter(inputValue))
+        if (this.props.eraseValue) {
+          this.elementInputRef.current.value = ''
+        }
+      } else {
+        this.setState({ error: new Error('The element must be unique and cannot be an empty string.') })
+      }
+    }
+  }
+
+  render() {
+    return (
+      <React.Fragment>
+        <input
+          autoFocus
+          className={classnames('form-control', this.state.error && style.inputError)}
+          ref={this.elementInputRef}
+          type="text"
+          placeholder={this.props.placeholder || ''}
+          defaultValue={this.props.defaultValue || ''}
```

Comment:
value and default represents the state of a component, since the component will play with them. This could be set in the didMount and didUpdate section. Also, the default value could be skipped altogether (this.setState(value = default value)